libs/api/domains/user-profile/src/lib/userProfile.service.ts

```diff
@@ -104,13 +122,40 @@ export class UserProfileService {
     const updateUserDto: UpdateUserProfileDto = {
       //temporary as schemas where not working properly
       locale: input.locale as string,
-      mobilePhoneNumber: input.mobilePhoneNumber,
-      email: input.email,
+
+      /* Mobile and email will be within islykill service */
+      // mobilePhoneNumber: input.mobilePhoneNumber,
+      // email: input.email,
     }
     const request: UserProfileControllerUpdateRequest = {
       nationalId: user.nationalId,
       updateUserProfileDto: updateUserDto,
     }
+
+    const feature = await this.featureFlagService.getValue(
+      Features.personalInformation,
+      false,
+      user,
+    )
+
+    if (!feature) {
+      handleError({ status: 'User profile update is feature flagged for user' })
+    }
+
+    const islyklarData = await this.islyklarService.getIslykillSettings(
+      user.nationalId,
+    )
+
+    // if email exists and *only* mobile is updated, email will become undefined. And vice-versa.
+    await this.islyklarService
+      .updateIslykillSettings(user.nationalId, {
+        email: input.email || islyklarData.email,
+        mobile: input.mobilePhoneNumber || islyklarData.mobile,
+        canNudge: input.canNudge,
+        bankInfo: input.bankInfo,
```

Comment:
Should `bankInfo` only be in update, not create?

Reply:
Yeah, so the bankInfo is not in create with the iskykill service. When a user is created the service will look for an existing bankInfo with Þjóðskrá. If it exists, the bankInfo will be updated "automatically".

If it does not exist, the user can edit on the "persónuupplýsingar" page.

While onboarding the user, the UI will only ask for email, phone + language.